next.config.js

```diff
@@ -27,6 +27,10 @@ const scheme = process.env.SCHEME === 'http' || process.env.SCHEME === 'https'
   ? process.env.SCHEME
   : 'https';
 
+const nextauthUrl = process.env.NEXTAUTH_URL === '' || process.env.NEXTAUTH_URL === 'null'
```

Comment:
Are you sure about testing against a string value of 'null' - normally someone is testing again the value null (not as a string)? I lately changed all this empty string plus null checks into simple checks just if the variable contains a value like this:

```
process.env.NEXTAUTH_URL  ?  `${scheme}://${process.env.NEXTAUTH_URL}` : process.env.VERCEL_URL ;
```